pyqtorch/modules/circuit.py

```diff
@@ -56,6 +72,12 @@ def _device(self) -> torch.device:
     def init_state(self, batch_size: int) -> torch.Tensor:
         return zero_state(self.n_qubits, batch_size, device=self._device)
 
+    def matrices(self, thetas: torch.Tensor) -> torch.Tensor:
+        matrix = self.operations[0].matrices(thetas[-1]).squeeze(2)
+        for i, operation in enumerate(self.operations[1:]):
+            matrix = matrix @ operation.matrices(thetas[-2 - i, :]).squeeze(2)
```

Comment:
This is only correct if all operations act on the same qubit index. I think we could make a separate PR for consistent matrix behavior of pyqtorch. There would not be a unitary hack reward on it but you could still work on it if you like?:)

Reply:
Indeed. I thought about that but forgot to do anything about it. I'm ok with making a separate PR later to fix that!

Reply:
also thought about this a bit more and added an issue here #71. we will have to refactor the other matrix methods for this as well.

Reply:
This method is not correct and should be removed:)